packages/rancher-wins-upgrader/charts/templates/daemonset.yaml

```diff
@@ -0,0 +1,45 @@
+{{- range .Values.winsConfigs }}
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: {{ template "winsUpgrader.name" $ }}-{{ .name }}
+  namespace: {{ template "winsUpgrader.namespace" $ }}
+  labels: {{ include "winsUpgrader.labels" $ | nindent 4 }}
+spec:
+  selector:
+    matchLabels: {{ include "winsUpgrader.labels" $ | nindent 6 }}
+  template:
+    metadata:
+      labels: {{ include "winsUpgrader.labels" $ | nindent 8 }}
+    spec:
+      nodeSelector: {{ include "winsUpgrader.nodeSelector" $ | nindent 8 }}
+{{- if .nodeSelector }}
+{{ toYaml .nodeSelector | indent 8 }}
+{{- end }}
+{{- if .tolerations }}
+      tolerations: {{ .tolerations | toYaml | nindent 8 }}
+{{- else }}
+      tolerations: {{ include "winsUpgrader.tolerations" $ | nindent 8 }}
+{{- end }}
+      serviceAccountName: {{ template "winsUpgrader.name" $ }}
+      containers:
+      - name: wins-upgrader
+        image: {{ template "system_default_registry" $ }}{{ .image.repository }}:{{ .image.tag }}
+        command: ["pwsh", "-f", "c:/scripts/upgrade.ps1"]
+        volumeMounts:
+        - name: wins
+          mountPath: c:/host/etc/rancher/wins
+        - name: upgrade-scripts
+          mountPath: c:/scripts/
+        env:
+        - name: HELM_REVISION_NUMBER
```

Comment:
This is a hack that forces helm to redeploy the DaemonSet on a helm upgrade since the revision number is always modified 😄